elastic/datadog_checks/elastic/elastic.py

```diff
@@ -802,22 +804,23 @@ def _process_metric(self, data, metric, xtype, path, xform=None,
             self._metric_not_found(metric, path)
 
     def _process_health_data(self, data, config):
-        if self.cluster_status.get(config.url) is None:
-            self.cluster_status[config.url] = data['status']
-            if data['status'] in ["yellow", "red"]:
-                event = self._create_event(data['status'], tags=config.tags)
+        status = data.get('status')
+        if not self.cluster_status.get(config.url):
+            self.cluster_status[config.url] = status
+            if status and status in ["yellow", "red"]:
+                event = self._create_event(status, tags=config.tags)
                 self.event(event)
 
-        if data['status'] != self.cluster_status.get(config.url):
-            self.cluster_status[config.url] = data['status']
-            event = self._create_event(data['status'], tags=config.tags)
+        if status != self.cluster_status.get(config.url):
+            self.cluster_status[config.url] = status
+            event = self._create_event(status, tags=config.tags)
             self.event(event)
 
         for metric, desc in self.CLUSTER_HEALTH_METRICS.iteritems():
             self._process_metric(data, metric, *desc, tags=config.tags)
 
         # Process the service check
-        cluster_status = data['status']
+        cluster_status = status
         if cluster_status == 'green':
             status = AgentCheck.OK
```

Comment:
Not a big fan of having the same variable name for the `service_check` status and the `elastic` status. Can you change one ?